ansible_ai_connect/main/ssl_manager.py

```diff
@@ -0,0 +1,181 @@
+            logger.debug("SSL Manager: Using REQUESTS_CA_BUNDLE: %s", self.requests_ca_bundle)
+            return self.requests_ca_bundle
+
+        # Priority 3: Service CA (for internal services)
+        if self.service_ca_path and os.path.exists(self.service_ca_path):
+            logger.debug("SSL Manager: Using service CA: %s", self.service_ca_path)
+            return self.service_ca_path
+
+        # Priority 4: Legacy settings service CA
+        if self._service_ca_path and os.path.exists(self._service_ca_path):
+            logger.debug("SSL Manager: Using legacy service CA: %s", self._service_ca_path)
+            return self._service_ca_path
+
+        # No custom CA bundle found - use system defaults
+        logger.debug("SSL Manager: No custom CA bundle found, using system defaults")
+        return None
+
+    def get_requests_session(self, verify_ssl: bool = True) -> requests.Session:
+        """Get requests session with infrastructure-managed SSL configuration.
+        Args:
+            verify_ssl: Whether SSL verification should be enabled
+        Returns:
+            Configured requests session with proper SSL settings.
+            Falls back to system defaults if custom CA bundles are not accessible.
+        """
+        session = requests.Session()
+
+        if verify_ssl:
+            try:
+                ca_bundle_path = self._get_ca_bundle_path()
+                if ca_bundle_path:
+                    # Verify the CA bundle file is readable before using it
+                    if os.path.exists(ca_bundle_path) and os.access(ca_bundle_path, os.R_OK):
+                        session.verify = ca_bundle_path
+                        logger.debug(
+                            "SSL Manager: Session configured with CA bundle: %s", ca_bundle_path
+                        )
+                    else:
+                        logger.warning(
+                            "SSL Manager: CA bundle not accessible: %s,"
+                            + " falling back to system defaults",
+                            ca_bundle_path,
+                        )
+                        # session.verify defaults to True, which uses system CAs
+                else:
+                    logger.debug("SSL Manager: Session using system default CA verification")
+                    # session.verify defaults to True, which uses system CAs
+
+            except (OSError, AttributeError) as e:
+                logger.warning(
+                    "SSL Manager: SSL configuration issue (non-fatal): %s, using system defaults", e
+                )
+                # Don't raise exception, just fall back to system defaults
+                # session.verify defaults to True
+        else:
+            session.verify = False
+            logger.debug("SSL Manager: SSL verification disabled")
+
+        return session
+
+    def get_ssl_context(self, verify_ssl: bool = True) -> Optional[ssl.SSLContext]:
```

Comment:
I don't see the value of the `verify_ssl` parameter considering you only test the value once and don't really use it. It would be more elegant to just remove the parameter and do the `verify_ssl` check before.